interfaces/acados_template/acados_template/acados_ocp_solver.py

```diff
@@ -258,6 +258,11 @@ def __init__(self, acados_ocp: Union[AcadosOcp, AcadosMultiphaseOcp], json_file=
         self.time_solution_sens_solve = 0.0
         self.time_solution_sens_lin = 0.0
 
+        # store nlp_solver_max_iter value
+        # (needed to raise a warning whenever max_iter is set to a higher value,
+        # since that could cause memory violations when writing to stats)
+        self.__nlp_solver_max_iter = self.solver_options['nlp_solver_max_iter']
```

Comment:
I would just compare to `self.solver_options['nlp_solver_max_iter']` directly below.
It would make sense to make `self.solver_options` private directly to not allow changing them from outside.

Reply:
> since that could cause memory violations when writing to stats

I think there will be no memory violations, it will just not store stats of iterations that exceed the initial number.
However, I don't think we should expose this behavior in any way.
So on a second thought, I would prefer raising an Exception below.

Reply:
I have made `solver_options` private 4cc423a7fb3 and now the code raises an Exception rather than printing a warning 5be544f80d14.

However, the tests are failing due to an error which doesn't seem to be related to the present MR:

`File "/home/runner/work/acados/acados/examples/acados_python/tests/reset_test.py", line 136, in main
    ocp_solver.set(i, 'x', np.NaN * np.ones((nx,)))
  File "/home/runner/work/acados/acados/acadosenv/lib/python3.10/site-packages/numpy/__init__.py", line 397, in __getattr__
    raise AttributeError(
AttributeError: `np.NaN` was removed in the NumPy 2.0 release. Use `np.nan` instead.. Did you mean: 'nan'?`